x86: add nforce eth to default packages

bobafetthotmail opened the pull request:
forcedeth is necessary to use the integrated
ethernet controller of Nvidia nForce chipset.

There are PC motherboards with this chipset
from 2001 that run 32bit Athlon XP CPUs and
more modern ones up to 2009 that can run Intel
and AMD 64bit processors, so add this to
all non-geode x86 targets.

Signed-off-by: Alberto Bursi <bobafetthotmail@gmail.com>

tested on a nForce motherboard with an Athlon XP, legacy target.

adrianschmutzler commented on target/linux/x86/64/target.mk:
```diff
@@ -1,6 +1,6 @@
 ARCH:=x86_64
 BOARDNAME:=x86_64
-DEFAULT_PACKAGES += kmod-button-hotplug kmod-e1000e kmod-e1000 kmod-r8169 kmod-igb kmod-bnx2
+DEFAULT_PACKAGES += kmod-button-hotplug kmod-e1000e kmod-e1000 kmod-r8169 kmod-igb kmod-bnx2 forcedeth
```
Since you are touching it, please add a line break here so one line is max. 80 characters.

bobafetthotmail committed:
x86: add nforce eth to default packages

forcedeth is necessary to use the integrated
ethernet controller of Nvidia nForce chipset.

There are PC motherboards with this chipset
from 2001 that run 32bit Athlon XP CPUs and
more modern ones up to 2009 that can run Intel
and AMD 64bit processors, so add this to
all non-geode x86 targets.

Signed-off-by: Alberto Bursi <bobafetthotmail@gmail.com>

bobafetthotmail commented:
Ok done. @adrianschmutzler 

While I was doing this change, I noticed something strange that could be cleaned up (as a separate PR). Is there a good reason why packages in x86 target are split (or even fully redundant as for geodes) between two different files?

I think this can be consolidated to be all defined either in one or in the other file, and the same type of file for all subtargets of x86. (i.e. so either the target.mk in the subtarget folder or the subtarget.mk files in the image folder)

I'm talking of these files:
target/linux/x86/64/target.mk and target/linux/x86/image/64.mk 
target/linux/x86/generic/target.mk and target/linux/x86/image/generic.mk 
target/linux/x86/legacy/target.mk and target/linux/x86/image/legacy.mk
target/linux/x86/geode/target.mk and target/linux/x86/image/geode.mk 

That may or may not have packages defined, or may both define the same package selection twice.

What do you think?

adrianschmutzler commented:
@bobafetthotmail In principle, I agree to you proposition. However, I have no experience with x86 target and thus cannot say whether this splitting of packages has a specific reason I'm not aware of.

hauke commented:
Thank you for your patch, it was applied to master.
